lib/resty/upstream/healthcheck.lua

```diff
@@ -524,12 +530,12 @@ function _M.spawn_checker(opts)
         return nil, "\"type\" option required"
     end
 
-    if typ ~= "http" then
+    if typ ~= "http" and typ ~= "tcp" then
```

Comment:
ran into this while browsing: updating this check also requires updating the error message below